Parser/asdl_c.py

```diff
@@ -723,8 +757,34 @@ def visitModule(self, mod):
                 goto cleanup;
         }
     }
+    PyObject *field, *field_default;
+    for (i = 0; i < numfields; i++) {
+        field = PySequence_GetItem(fields, i);
+        if (!field) {
+            res = -1;
+            goto cleanup;
+        }
+        int attr_present = PyObject_HasAttr(self, field);
+        Py_DECREF(field);
+        if (!attr_present) {
+            field_default = PySequence_GetItem(field_defaults, i);
+            if (!field_default) {
+                res = -1;
+                goto cleanup;
+            }
+            if (!(field_default = find_field_default(field_default))) {
```

Comment:
Please, do not reasign in this way, this made this code very difficult to read and to track the references and what is being decremented and what not. Use a new variable and decrement `field_default` here and not on the `find_field_default` function.

Reply:
I see, will do.

Reply:
As an example of why this was more difficult to read, that is why I was confused with https://github.com/python/cpython/pull/19031#discussion_r393286073